spec.md

```diff
@@ -0,0 +1,956 @@
+
+> [!NOTE]
+>
+> The `START` and `END` values do not affect the placement of notes nor any other time codes relative to the audio.
+> They simply indicate that a song should be started or stopped a certain amount of time into the audio file.
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> In versions 0.x and 1.x `START` was specified in seconds as a decimal value.
+> Version 2.0.0 changes this to an integer and milliseconds.
+
+### 3.15. The `PREVIEWSTART` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        0.2.0
+```
+
+The `PREVIEWSTART` header indicates a time offset in milliseconds relative to the start of the audio where the preview starts.
+Implementations MAY use this value when playing a song in a preview setting (e.g. during song selection).
+In its absence implementations SHOULD default to the start of the medley section (if available).
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> In versions 0.x and 1.x `PREVIEWSTART` was specified in seconds as a decimal value.
+> Version 2.0.0 of this specification changes this to an integer and milliseconds.
+
+### 3.16. The `MEDLEYSTART` and `MEDLEYEND` Headers
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        2.0.0
+```
+
+The `MEDLEYSTART` and `MEDLEYEND` headers indicate in milliseconds the start and end of the medley section of a song relative to the start of the audio.
+These tags replace `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` in version 2.0.0 of this specification.
+
+### 3.17. The `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` Headers
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        0.2.0
+Removed:      2.0.0
+```
+
+The `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` headers indicate in beats the start and end of the medley section of a song.
+Implementations MUST respect the `GAP` value when calculating the medley start and end times.
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> The headers `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` have been replaced by `MEDLEYSTART` and `MEDLEYEND` in version 2.0.0 of this specification.
+
+### 3.18. The `CALCMEDLEY` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       "on" / "off"
+Since:        0.2.0
+```
+
+If [`MEDLEYSTART`](#316-the-medleystart-and-medleyend-headers) or [`MEDLEYEND`](#316-the-medleystart-and-medleyend-headers) ([`MEDLEYSTARTBEAT`](#317-the-medleystartbeat-and-medleyendbeat-headers) or [`MEDLEYENDBEAT`](#317-the-medleystartbeat-and-medleyendbeat-headers) for versions before 2.0.0) are not specified,
+the `CALCMEDLEY` header indicates whether an implementation is supposed to determine the start and end of the medley section automatically.
+The value of this header is compared case-insensitively.
+
+> [!CAUTION]
+>
+> The exact semantics of the `CALCMEDLEY` header have not been defined yet.
+
+### 3.19. The `YEAR` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       4*DIGIT
```

Comment:
```suggestion
Syntax:       4DIGIT
```

Reply:
I have changed this both ways now. It seems there is a discussion to be had about the syntax we want. I have changed it according to your suggestion (because I feel it best reflects the syntax outlined by the [website](http://usdx.eu/format/)). I also added a caution about the exact syntax.

Reply:
What's to be decided about the #YEAR header?

Reply:
The way I'm currently reading the website the `YEAR` header has to consist of exactly 4 digits. So a spec-compliant implementation would have to reject a song with `#YEAR:123`, because that's invalid syntax.

@Baklap4 suggested that the syntax should be `4*DIGIT`, i.e. (positive) number with at least 4 digits. So `#YEAR:123` would still be invalid, but `#YEAR:123456` would be allowed.

I personally think that artificially restricting the range of allowed years serves no real purpose in the spec and would like to allow any integer (positive or negative). That also simplifies parsing, because most programming languages have some sort of `parseInt` function that handles any integer.

Because this PR and the discussion is long enough as it is I feel like this warrants its own issue for discussion.

Reply:
Not sure if I can follow. I think any non-four-digit year makes no sense, so I would reject (read: ignore) any year value that is anything else but a four digit integer (we could even be more restrictive and restrict the first digit to be either a 1 or a 2).

Reply:
I can definitely understand that perspective. I have some more thoughts on this but I think I'll hold on to those for now and open a follow-up issue later on (and leave the caution marker for now). Otherwise this PR might get more unwieldy than it already is.